### ISSUE TITLE
Support `wrapt`

### ISSUE THREAD

**Issue by dimaqq** [microsoft/pyright#990]
`wrapt` creates decorators via object proxies.
(https://github.com/GrahamDumpleton/wrapt)

Today, `pyright` completely balks at functions decorated with wrapt.

Given the canonical example:

```py
import wrapt

@wrapt.decorator
def pass_through(wrapped, instance, args, kwargs):
    return wrapped(*args, **kwargs)

@pass_through
def function():
    pass
```

`pyright` says that `Arguments missing for parameters "instance", "args", "kwargs"` 🙀

**Comment by erictraut** [comment]
This library is performing dynamic behaviors that can't be described using normal Python type annotations, nor is it feasible for a static type checker to infer the types in this case. This behavior could could conceivably be described through the use of custom logic in a type checker, but Pyright doesn't include such custom logic for third-party libraries. We may eventually develop a plugin model that allows extensions that describe such behaviors, but this is unlikely to happen any time soon.

If you want to use static type checking, I recommend avoiding libraries that exhibit non-standard behaviors that cannot be described using type annotations. 

You can, of course, suppress diagnostic messages on a per-line, per-file or per-rule basis.
